tripy/tripy/frontend/trace/trace.py

```diff
@@ -120,6 +120,11 @@ def to_flat_ir(self):
         from tripy.flat_ir.flat_ir import FlatIR
 
         flat_ir = FlatIR(shapes=self.shapes)
+        # Assign shapes to static shape arguments to ease translation and optimizations during the lowering to MLIR.
+        if self.shapes:
+            for input, shape_bounds in zip(self.inputs, self.shapes):
+                if shape_bounds.is_static():
+                    input.shape = shape_bounds.min
```

Comment:
shall we check that `input.shape` matches `shape_bounds.min` unless it is `-1`?

Reply:
`shape_bounds` are populated by `InputInfo` and will not be `-1`. Will add an assert.

Reply:
I was also wondering if we need to check `input.shape` is not overridden by `shape_bounds.min` if it is not -1.

Reply:
Added a check for `input.shape is None`.